app/frontend/src/features/profile/edit/EditProfile.tsx

```diff
@@ -368,15 +368,20 @@ export default function EditProfileForm() {
             {languages && (
               <Controller
                 control={control}
-                defaultValue={user.languageAbilitiesList.map(
-                  (ability) => languages[ability.code]
+                defaultValue={user.languageAbilitiesList.map((ability) =>
+                  languages.get(ability.code)
                 )}
                 name="fluentLanguages"
                 render={({ onChange, value }) => (
                   <ProfileTagInput
                     onChange={(_, value) => onChange(value)}
                     value={value}
-                    options={Object.values(languages)}
+                    getOptionLabel={(languageCode) =>
+                      languages.get(languageCode)!
+                    }
+                    options={Array.from(
+                      languages.keys() as IterableIterator<string>
```

Comment:
Not sure if this is the culprit... but without this type assertion, the types of `Array.from` won't accept this as a valid iterator, probably because Google rolled their crappier version with the generated code and it's missing something. But it still seems to work at first until you go to the route directly...